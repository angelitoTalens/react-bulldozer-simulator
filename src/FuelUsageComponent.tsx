import React from "react";
import "./styles/FuelUsageComponent.scss"
import Container from "react-bootstrap/Container";
import Table from "react-bootstrap/Table";
import { Cost } from "./EnumTypes";


export type FuelUsageComponentProps = {
     clearPlainLand: Cost;
     clearRockyLand: Cost;
     clearTreePlantedLand: Cost;
     revisitClearedLand: Cost;
}

function FuelUsageComponent(props: FuelUsageComponentProps) {
    const totalVisit = props.clearPlainLand.quantity +
                       props.clearRockyLand.quantity +
                       props.clearTreePlantedLand.quantity +
                       props.revisitClearedLand.quantity;
    
    const totalFuelUsage = props.clearPlainLand.value +
                           props.clearRockyLand.value +
                           props.clearTreePlantedLand.value +
                           props.revisitClearedLand.value;
    
    return (
        <Container className="fuel-usage-container">
            <Table size="sm" striped hover>
                <thead>
                    <tr>
                        <th style={{width: "40%"}}>Activity</th>
                        <th style={{width: "30%"}}>Visits</th>
                        <th style={{width: "30%"}}>Fuel Usage (fuel units)</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td>Clearing Plain Land</td>
                        <td>{props.clearPlainLand.quantity}</td>
                        <td>{props.clearPlainLand.value}</td>
                    </tr>
                    <tr>
                        <td>Clearing Rocky Land</td>
                        <td>{props.clearRockyLand.quantity}</td>
                        <td>{props.clearRockyLand.value}</td>
                    </tr>
                    <tr>
                        <td>Clearing Tree Planted Land</td>
                        <td>{props.clearTreePlantedLand.quantity}</td>
                        <td>{props.clearTreePlantedLand.value}</td>
                    </tr>
                    <tr>
                        <td>Revisiting Cleared Land</td>
                        <td>{props.revisitClearedLand.quantity}</td>
                        <td>{props.revisitClearedLand.value}</td>
                    </tr>
                    <tr>
                        <td><b>Total</b></td>
                        <td><b>{totalVisit}</b></td>
                        <td><b>{totalFuelUsage}</b></td>
                    </tr>
                </tbody>
            </Table>
        </Container>
    );
}

export default FuelUsageComponent;