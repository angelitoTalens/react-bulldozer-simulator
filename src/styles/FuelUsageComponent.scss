.fuel-usage-container {
    padding-top: 50px;
}