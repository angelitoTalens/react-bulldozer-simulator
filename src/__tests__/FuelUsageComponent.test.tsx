import React from 'react';
import renderer from 'react-test-renderer';

import FuelUsageComponent from '../FuelUsageComponent';

test ("Test FileInputComponent", ()  => {
    const component = renderer.create(
        <FuelUsageComponent
            clearPlainLand={{quantity: 1, value: 10}}
            clearRockyLand={{quantity: 1, value: 20}}
            clearTreePlantedLand={{quantity: 1, value: 20}}
            revisitClearedLand={{quantity: 1, value: 10}}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});