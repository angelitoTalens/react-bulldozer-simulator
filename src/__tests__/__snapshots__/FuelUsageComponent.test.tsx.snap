// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FileInputComponent 1`] = `
<div
  className="fuel-usage-container container"
>
  <table
    className="table table-sm table-striped table-hover"
  >
    <thead>
      <tr>
        <th
          style={
            Object {
              "width": "40%",
            }
          }
        >
          Activity
        </th>
        <th
          style={
            Object {
              "width": "30%",
            }
          }
        >
          Visits
        </th>
        <th
          style={
            Object {
              "width": "30%",
            }
          }
        >
          Fuel Usage (fuel units)
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Clearing Plain Land
        </td>
        <td>
          1
        </td>
        <td>
          10
        </td>
      </tr>
      <tr>
        <td>
          Clearing Rocky Land
        </td>
        <td>
          1
        </td>
        <td>
          20
        </td>
      </tr>
      <tr>
        <td>
          Clearing Tree Planted Land
        </td>
        <td>
          1
        </td>
        <td>
          20
        </td>
      </tr>
      <tr>
        <td>
          Revisiting Cleared Land
        </td>
        <td>
          1
        </td>
        <td>
          10
        </td>
      </tr>
      <tr>
        <td>
          <b>
            Total
          </b>
        </td>
        <td>
          <b>
            4
          </b>
        </td>
        <td>
          <b>
            60
          </b>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
